lib/main.js

```diff
@@ -55,14 +67,56 @@ export default {
           });
         }
       }),
-    );
-
-    this.subscriptions.add(
       atom.config.observe('linter-tslint.useLocalTslint', (use) => {
         tslintCache.clear();
         this.useLocalTslint = use;
       }),
     );
+
+    atom.workspace.observeActivePaneItem((activeItem) => {
+      if (activeItem && activeItem.getGrammar) {
+        if (isTypescriptGrammar(activeItem.getGrammar())) {
+          activeItem.element.classList.add(editorClass);
+        }
+      }
+    });
+
+    // Command subscriptions
+    this.subscriptions.add(
+      atom.commands.add(`atom-text-editor.${editorClass}`, {
+        'linter-tslint:fix-file': () => {
+          const textEditor = atom.workspace.getActiveTextEditor();
+          // const filePath = textEditor.getPath();
+          // const text = textEditor.getText();
+
+          if (!textEditor || textEditor.isModified()) {
+            // Abort for invalid or unsaved text editors
+            atom.notifications.addError('Linter-TSLint: Please save before fixing');
+            return Promise.resolve();
```

Comment:
I though async/await didn't make the ES6 spec and wasn't going to have wide support until ECMA 2017. I will change and test it out though